### PR TITLE
tl;dr modded sinks are actually practical now

### PR DESCRIPTION
**Pull request by ThomasSpeedrunner**
You may now use countsAsWaterSource, countsAsLavaSource, and countsAsHoneySource to easily make any tile into a source for the vanilla liquid of your choice --- in other words, you no longer need to IL edit FindRecipes to make that happen.

Modded liquids sold separately.

### CODE REVIEW

**Comment by Chicken-Bones** [comment]
This will consume a lot of memory. Please use a `TileID.Sets` entry instead

**Comment by ThomasSpeedrunner** [comment]
should be better now

File: patches/tModLoader/Terraria/WorldGen.cs.patch [hunk 1]
```diff
@@ -975,7 +975,14 @@
  			for (int num7 = num - 1; num7 < num + 3; num7++) {
  				for (int num8 = num2 - 1; num8 < num2 + 3; num8++) {
  					TileFrame(num7, num8);
-@@ -31181,9 +_,6 @@
+@@ -31176,14 +_,13 @@
+ 					tileSafely.frameX = (short)(k * 18);
+ 					tileSafely.frameY = (short)(style * num + l * 18);
+ 					tileSafely.type = type;
++					if (type == 172)
++						tileSafely.countsAsWaterSource = true;
```
Comment (JavidPack) [inline patches/tModLoader/Terraria/WorldGen.cs.patch:983]: I think this edit wasn't reverted from the old approach

File: patches/tModLoader/Terraria/ID/TileID.cs.patch [hunk 2]
```diff
@@ -23,3 +23,13 @@
  			public static bool[] AllowsSaveCompressionBatching = Factory.CreateBoolSet(true, 520, 423);
  			public static bool[] IsATreeTrunk = Factory.CreateBoolSet(false, 5, 72, 583, 584, 585, 586, 587, 588, 589, 596, 616);
  			public static bool[] IsShakeable = Factory.CreateBoolSet(false, 5, 72, 323, 583, 584, 585, 586, 587, 588, 589, 596, 616);
+@@ -196,6 +_,9 @@
+ 			public static bool[] IgnoredInHouseScore = Factory.CreateBoolSet(4, 3, 73, 82, 83, 84);
+ 			public static bool[] SpreadOverground = Factory.CreateBoolSet(2, 23, 32, 60, 70, 109, 199, 352, 477, 492);
+ 			public static bool[] SpreadUnderground = Factory.CreateBoolSet(23, 109, 199, 60, 70);
++			public static bool[] CountsAsWaterSource = Factory.CreateBoolSet(172);
```
Comment (Mirsario) [inline patches/tModLoader/Terraria/ID/TileID.cs.patch:30]: Use X.TML.cs partial files for additions.

File: ExampleMod/Content/Tiles/ExampleSink.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.ObjectData;
+
+namespace ExampleMod.Content.Tiles
+{
+	public class ExampleSink : ModTile
+	{
+		public override void SetDefaults()
+		{
```
Comment (Mirsario) [inline ExampleMod/Content/Tiles/ExampleSink.cs:12]: This and the other example have the wrong brace style. Make your editor use the provided editorconfig.

**Review by Mirsario — CHANGES_REQUESTED** [review]

File: patches/tModLoader/Terraria/WorldGen.cs.patch (hunk 1 above)
Comment (ThomasSpeedrunner) [reply patches/tModLoader/Terraria/WorldGen.cs.patch]: fixed

File: patches/tModLoader/Terraria/ID/TileID.cs.patch (hunk 2 above)
Comment (ThomasSpeedrunner) [reply patches/tModLoader/Terraria/ID/TileID.cs.patch]: done

File: ExampleMod/Content/Tiles/ExampleSink.cs (hunk 3 above)
Comment (ThomasSpeedrunner) [reply ExampleMod/Content/Tiles/ExampleSink.cs]: fixed

**Comment by ThomasSpeedrunner** [comment]
anything else?

**Review by Mirsario — APPROVED** [review]

**Pull request merged**